### DISCUSSION TITLE
Include memberdata on API request on no door access

### DISCUSSION THREAD

**Discussion by olmari** [TampereHacklab/mulysa#382] [Ideas]
Currently Mulysa gives response body with member data for door query only when member has valid door access. For Vaasa Gatekeeper to report meaningful data in staff room, add response body for known members without door access too (currently HTTP 481 response).

Additionally we could define more codes for various states of member status, for example in style: Kuukausimaksu overdue == 283, Kuukausimaksu suspended == 483, etc. This way staff would know immediately who called to door, payment status and why it possibly failed.

**Comment by vranki** [comment]
I don't think response codes is way to go - you would have to do some kind of site-specific mapping file between each status and a code for it. Adding more data to response would be my suggestion.

**Comment by olmari** [comment]
That's why response codes exist though.

**Comment by olmari** [comment]
Improved an bit with response body in member but access denied in https://github.com/TampereHacklab/mulysa/pull/297
